### PR TITLE
Support of correct_umis for a pre-defined set of barcodes

### PR DESCRIPTION
**Pull request by naumenko-sa**

### CODE REVIEW

File: docs/contents/code.rst [hunk 1]
```diff
@@ -84,29 +84,61 @@ The most useful modules inside ``bcbio``, ordered by likely interest:
...
 
-For developing and testing changes locally, you can install directly into a
-bcbio-nextgen installation. The automated bcbio-nextgen
-installer creates an isolated Python environment using `Anaconda`_. This will be
-a subdirectory of your installation root, like
-``/usr/local/share/bcbio-nextgen/anaconda``. The installer also includes a
-``bcbio_python`` executable target which is the python in this isolated anaconda
-directory. You generally will want to make changes to your local copy of the
-bcbio-nextgen code and then install these into the code directory.
+Creating a separate bcbio installation
```
Comment (lbeltrame) [inline docs/contents/code.rst]: These are unrelated changes as well.

File: docs/contents/code.rst [hunk 2]
```diff
@@ -84,29 +84,61 @@ The most useful modules inside ``bcbio``, ordered by likely interest:
 Development infrastructure
 ==========================
 
+GitHub
```
Comment (lbeltrame) [inline docs/contents/code.rst]: Unrelated change to this patch.

File: docs/contents/configuration.rst [hunk 3]
```diff
@@ -738,9 +738,10 @@ Somatic variant calling
 =======================
 
 - ``min_allele_fraction`` Minimum allele fraction to detect variants in
-  heterogeneous tumor samples, set as the float or integer percentage to
+  heterogeneous tumor samples, set as the float or integer **percentage** to
```
Comment (lbeltrame) [inline docs/contents/configuration.rst]: Unrelated.

File: docs/contents/configuration.rst [hunk 4]
```diff
@@ -942,6 +943,11 @@ Configuration options for UMIs:
   ``fastq_name`` with UMIs in read names or the path to a fastq file with
   UMIs for each aligned read.
 
+- ``correct_umis: [path/to/whitelist_umi.txt]``. For a restricted set of UMIs specify a
```
Comment (lbeltrame) [inline docs/contents/configuration.rst]: Can this be expanded a bit? What is exactly a "restricted" set of UMIs? Even the Fulcrum Genomics page isn't too clear.

**Comment by lbeltrame** [comment]
This PR mixes two different changes, I would suggest to split the doc changes and submit them separately.

**Comment by roryk** [comment]
Thanks Sergey and Luca. I squashed this and annotated the changes in the commit; the extras were just documentation fixes so it's cool for them to hitch a ride on the feature commit. Thanks a lot Sergey for implementing this!
